controllers/reportTagController.js

```diff
@@ -0,0 +1,105 @@
+const {
+    createReportTag,
+    getAllReportTags,
+    deleteTag
+  } = require('../lib/modelManagers/reportTagModel'),
+  {
+    failureResponse,
+    successResponse
+  } = require('../lib/utils/helper_function'),
+  {
+    OK_CODE,
+    RESOURCE_CREATED_CODE,
+    REPORT_TAG_CREATED,
+    NO_TAG_FOUND,
+    RETRIEVED_REPORT_TAG,
+    SERVER_ERROR_CODE,
+    TAG_DELETED,
+    SERVER_ERROR_MESSAGE,
+    SEQUELIZE_UNIQUE_ERROR,
+    CONFLICT_CODE,
+    TAG_ALREADY_EXIST
+  } = require('../constants/index'),
+  pagination = require('../lib/utils/pagination/paginationHelper');
+/**
+ * @description - handle report tag
+ *
+ * @class ReportTagController
+ */
+class ReportTagController {
+  /**
+   * @description - handle creating of report tag
+   *
+   * @static
+   * @param {any} req - Request Object
+   * @param {any} res - Response Object
+   * @returns {object} Returns the created tag
+   *
+   * @memberOf ReportTagController
+   */
+  static async createTag(req, res) {
+    try {
+      const { name } = req.body;
+      const createdTag = await createReportTag(name);
+      if (createdTag) {
+        return successResponse(
+          res,
+          REPORT_TAG_CREATED,
+          createdTag,
+          RESOURCE_CREATED_CODE
+        );
+      }
+    } catch (error) {
+      if (error.name === SEQUELIZE_UNIQUE_ERROR) {
+        return failureResponse(res, TAG_ALREADY_EXIST, CONFLICT_CODE);
+      }
+      return failureResponse(res, SERVER_ERROR_MESSAGE, SERVER_ERROR_CODE);
+    }
+  }
+
+  /**
+   * @description - handle retriving all report tags
+   *
+   * @static
+   * @param {any} req Request object
+   * @param {any} res Response obhect
+   * @returns {object} Returns object of tags
+   * @memberOf ReportTagController
+   */
+  static async getTags(req, res) {
+    const { limit, offset } = pagination(req.query);
+    try {
+      const tags = await getAllReportTags(limit, offset);
+      if (tags.length > 0) {
+        return successResponse(res, RETRIEVED_REPORT_TAG, tags, OK_CODE);
+      }
+      return successResponse(res, NO_TAG_FOUND, tags, OK_CODE);
+    } catch (error) {
+      return failureResponse(res, SERVER_ERROR_MESSAGE, SERVER_ERROR_CODE);
+    }
+  }
+
+  /**
+ * @description - handle deleting report tag
+ *
+ * @static
+ * @param {any} req
+ * @param {any} res
+ * @returns {null} no return
+ *
+ * @memberOf ReportTagController
+ */
+  static async deleteReportTag(req, res) {
+    const { tagId } = req.params;
+    try {
+      const deletedTag = await deleteTag(tagId);
+      if (deletedTag) {
+        return successResponse(res, TAG_DELETED, deleteTag, OK_CODE);
+      }
```

Comment:
what happens when no report was deleted?

Reply:
It will return an error as seen in the screenshot below. Because there is a middleware that validates the UUID.

<img width="948" alt="screenshot 2019-02-05 at 12 41 08 pm" src="https://user-images.githubusercontent.com/13318873/52271285-7ee51200-2943-11e9-9d66-d30429bf4f00.png">